test/extended/cli/explain.go

```diff
@@ -498,12 +498,37 @@ var (
 	}
 )
 
+func remove(slice []schema.GroupVersionResource, s int) []schema.GroupVersionResource {
+	if s > 0 || s > len(slice)-1 {
+		return slice
+	}
+	return append(slice[:s], slice[s+1:]...)
+}
+
 func getCrdTypes(oc *exutil.CLI) []schema.GroupVersionResource {
 	isMicroShift, err := exutil.IsMicroShiftCluster(oc.AdminKubeClient())
 	o.Expect(err).NotTo(o.HaveOccurred())
 	if isMicroShift {
 		return microshiftCRDTypes
 	}
+	isHypershift, _ := exutil.IsHypershift(context.TODO(), oc.AdminConfigClient())
+	if isHypershift {
+		etcdsResourceIndex := -1
+		for i := 0; i < len(baseCRDTypes); i++ {
+			if baseCRDTypes[i].Group != "operator.openshift.io" {
+				continue
+			}
+			if baseCRDTypes[i].Version != "v1" {
+				continue
+			}
+			if baseCRDTypes[i].Resource != "etcds" {
+				continue
+			}
+			etcdsResourceIndex = i
+			break
+		}
+		remove(baseCRDTypes, etcdsResourceIndex)
```

Comment:
Based on how the rest of the code is structured, it seems like etcd should be removed from the baseCRDTypes and to its own type that is added for non-HyperShift.  Although this is a fairly significant refactor and I'd like Red Hat's opinion before asking you to make this change.